test/notebooks/test_notebooks.py

```diff
@@ -120,9 +120,10 @@ def test_fl_with_trainconfig(isolated_filesystem, start_remote_server_worker_onl
 
 def test_fl_sms(isolated_filesystem):
     sys.path.append("advanced/Federated SMS Spam prediction/")
-    os.chdir("advanced/Federated SMS Spam prediction/")
```

Comment:
I think the best solution would be not to call ```os.chdir``` at all. I see [here](https://stackoverflow.com/questions/23619595/pythons-os-chdir-function-isnt-working) - the link you provided - that it has different behaviors on different operating systems.
What I am thinking is to add the ```preprocess.py``` path to ```sys``` and then simply run the notebook (how it is done now).

This will mean that on line 137, notebook will be ```p_name / notebook```.

What do you think?

Reply:
Just for the record... Tried this with @gmuraru but it doesn't work so the best solution is the current one on the PR.